Managing multiple redirects in Qwik (#6909)
QwikDev/qwik: packages/docs/src/routes/docs/(qwikcity)/guides/redirects/index.mdx

If you do not provide a status code, Qwik City will default to a `302` Found status.

Read more about redirect status codes [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages).

## Managing multiple redirects

In some cases, you may need to manage multiple redirects based on different conditions. For example, you might want to redirect users from old URLs to new URLs after a site restructure. Additionally, you may want editors in a CMS to manage these redirects as well. Here's one of the ways you can handle multiple redirects in Qwik:

```tsx title="src/routes/layout.tsx"
import { type RequestHandler } from "@builder.io/qwik-city";

export const onGet: RequestHandler = async ({ url, redirect }) => {
  // qwik city request caching ...

  // example external data source

>   async function fetchRules(): Promise<
>     { source: string; destination: string; permanent: boolean }[]
>   > {
>     // Simulating fetched data
>     return [
>       { source: "/old-path", destination: "/new-path", permanent: true },
>       {
>         source: "/another-old-path",
>         destination: "/another-new-path",
>         permanent: false,
>       },
>       // Add more rules as needed. These properties above can be data grabbed from a CMS or API.
>     ];
>   }
> 
>   const redirectData = fetchRules().then(
>     (results) =>
>       results
>         .filter((content) => {
>           return !!(content.source && content.destination);
>         })
>         // Create the "redirect" object Qwik expects
>         .map((data) => ({
>           source: data.source,
>           destination: data.destination,
>           permanent: !!data.permanent,
>         })),
>     (error) => {
>       // Logs issues with redirects without failing the deployment.
>       // Remove this function to stop deployment on redirect issues.
>       console.error("Error setting up redirects", error);
>       return [];
>     },
>   );
> 
>   const redirectRules = await redirectData;

Reviewer: ```suggestion
  async function fetchRules(): Promise<
    { source: string; destination: string; permanent: boolean }[]
  > {
    // Fetch data from a CMS or API, and add more rules as needed.
    // Filter and map your data to make it easier to handle it, as simulated here:
    return [
      { source: "/old-path", destination: "/new-path", permanent: true },
      {
        source: "/another-old-path",
        destination: "/another-new-path",
        permanent: false,
      },
    ];
  }

  const redirectRules = await fetchRules();
```
Does it really add value to the example to filter/map rules here after we fetch the rules? Especially considering that we simulate the fetched rules with only valid data.

As the goal of this example seems to be only to demonstrate how to handle multiple redirects, I would skip that part and consider `fetchRules` only returns valid data.

Though we could add a comment inside `fetchRules` to invite the readers to get & filter/rules their own properly.
